File: hexchat.wp.mpd.py
import subprocess

import hexchat

__module_name__ = 'MPD'
__module_version__ = '0.1.0'
__module_description__ = 'MPD currently playing script'


def now_playing(*dummy):
    proc = subprocess.check_output(['mpc',
                                    'current', '-f', '%artist% - %title% 「%album%」'])
    hexchat.command('SAY np: ' + proc.decode('utf-8').strip())
    return hexchat.EAT_ALL


def unload_callback(dummy):
    hexchat.prnt('RIP MPD wp script')

hexchat.hook_command('wp', now_playing, help='"/wp" to display currently playing MPD track')
hexchat.hook_unload(unload_callback)
hexchat.prnt('hexchat.wp.MPD.py loaded')

File: hexchat.wp.mpd.lua
hexchat.register('MPD', '1', 'MPD currently playing script')

local function get_np()
	local handle = assert(io.popen(
			'mpc ' ..
			'current ' ..
			'-h /var/lib/mpd/socket ' ..
			'-f "%artist% - %title% 「%album%」"'
	))
	local res = string.gsub(assert(handle:read('*a')), '\n', '')
	handle:close()
	hexchat.command('SAY np: ' .. res)
	return hexchat.EAT_ALL
end

hexchat.hook_command('wp', get_np, '"/wp" to display currently playing MPD track')
hexchat.print('hexchat.wp.mpd.lua loaded')
